tests/test_pipedrive_bookmarks.py

```diff
@@ -0,0 +1,162 @@
+        second_sync_bookmarks = menagerie.get_state(conn_id)
+
+        for stream in expected_streams:            
+            with self.subTest(stream=stream):
+
+                # expected values
+                expected_replication_method = expected_replication_methods[stream]
+
+                # collect information for assertions from syncs 1 & 2 base on expected values
+                first_sync_count = first_sync_record_count.get(stream, 0)
+                second_sync_count = second_sync_record_count.get(stream, 0)
+                first_sync_messages = [record.get('data') for record in
+                                       first_sync_records.get(
+                                           stream, {}).get('messages', [])
+                                       if record.get('action') == 'upsert']
+                second_sync_messages = [record.get('data') for record in
+                                        second_sync_records.get(
+                                            stream, {}).get('messages', [])
+                                        if record.get('action') == 'upsert']
+                first_bookmark_key_value = first_sync_bookmarks.get('bookmarks', {stream: None}).get(stream)
+                second_bookmark_key_value = second_sync_bookmarks.get('bookmarks', {stream: None}).get(stream)
+                
+
+                if expected_replication_method == self.INCREMENTAL:
+
+                    # collect information specific to incremental streams from syncs 1 & 2
+                    replication_key = list(expected_replication_keys[stream])[0] #Key in which state has been saved in state file
+                    record_replication_key = list(expected_replication_keys[stream])[0]
+
+                    first_bookmark_value = first_bookmark_key_value.get(replication_key)
+                    second_bookmark_value = second_bookmark_key_value.get(replication_key)
+                    
+                    first_bookmark_value_ts = self.dt_to_ts(first_bookmark_value)
+                    
+                    second_bookmark_value_ts = self.dt_to_ts(second_bookmark_value)
+                    
+                    simulated_bookmark_value = self.dt_to_ts(new_state['bookmarks'][stream][replication_key])
+
+                    # Verify the first sync sets a bookmark of the expected form
+                    self.assertIsNotNone(first_bookmark_key_value)
+                    self.assertIsNotNone(first_bookmark_value)
+                    
+                    self.assertIsNotNone(second_bookmark_key_value)
+                    self.assertIsNotNone(second_bookmark_value)
+
+                    # Verify the second sync bookmark is Equal to the first sync bookmark
+                    # assumes no changes to data during test
+                    # The dealflow stream is writing bookmark as a 'current_time-3_hours' so 1st and 2nd bookmark will be different
```

Comment:
Changing the simulated bookmark would not cause the final bookmark in sync 2 to be different. This could occur if the data is changing during the test, or if the tap is not saving state based on the max replication key value. The root cause should be identified.

Reply:
As per our discussion in [slack](https://talend.slack.com/archives/C01TGC4QAUE/p1653569610908509?thread_ts=1653565178.044539&cid=C01TGC4QAUE). We created a card [TDL-19236](https://jira.talendforge.org/browse/TDL-19236)  for the bookmarking suggestion and raised #117  for the same.